### PR TITLE
issue: 840733 Avoid using mce_sys_var in fast path

### PR DESCRIPTION
**Pull request by liranoz12**
Signed-off-by: Liran Oz <lirano@mellanox.com>

### CODE REVIEW

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/accl-libvma-pr/275/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test PASSed.
See http://bgate.mellanox.com/jenkins/job/accl-libvma-pr/256/ for details (Mellanox internal link).

File: src/vma/proto/dst_entry_udp.h [hunk 1]
```diff
@@ -56,6 +56,13 @@ class dst_entry_udp : public dst_entry
 	virtual ssize_t 	pass_buff_to_neigh(const iovec *p_iov, size_t & sz_iov, uint16_t packet_id = 0);
 	atomic_t m_a_tx_ip_id;
 	size_t m_n_tx_ip_id;
+
+private:
+
```
Comment (rosenbaumalex) [inline src/vma/proto/dst_entry_udp.h]: formatting: please remove extra line

File: src/vma/proto/neighbour.h [hunk 2]
```diff
@@ -350,6 +350,10 @@ class neigh_entry : public cache_entry_subject<neigh_key, neigh_val *>, public e
 private:
 	bool 			m_is_first_send_arp;
 
+	const uint32_t	m_n_neigh_wait_till_send_arp_msec;
```
Comment (rosenbaumalex) [inline src/vma/proto/neighbour.h]: formatting: please remove extra line and align member with space/tab to other memebrs

**Comment by mellanox-github** [comment]
Test PASSed.
See http://bgate.mellanox.com/jenkins/job/accl-libvma-pr/258/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/accl-libvma-pr/277/ for details (Mellanox internal link).

**Comment by rosenbaumalex** [comment]
@liranoz12 please check the merge conflicts and update PR

**Comment by mellanox-github** [comment]
Test PASSed.
See http://bgate.mellanox.com/jenkins/job/accl-libvma-pr/263/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/accl-libvma-pr/282/ for details (Mellanox internal link).

**Comment by OphirMunk** [comment]
For each m_XXX member which replaces safe_mce_sys().XXX need to:
1. Keep the same type definition
2. Use the same casting in code

**Comment by igor-ivanov** [comment]
bot:retest

**Comment by mellanox-github** [comment]
Test PASSed.
See http://bgate.mellanox.com/jenkins/job/accl-libvma-pr/269/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/accl-libvma-pr/288/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test PASSed.
See http://bgate.mellanox.com/jenkins/job/accl-libvma-pr/280/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/accl-libvma-pr/298/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test PASSed.
See http://bgate.mellanox.com/jenkins/job/accl-libvma-pr/285/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/accl-libvma-pr/303/ for details (Mellanox internal link).

**Comment by OphirMunk** [comment]
:+1: 
Please consider the following commit message
```
issue: 840733 Avoid using mce_sys variable in fast path

Save members of global struct safe_mce_sys().<member> in relevant instance members
m_sysvar_<member>. Accessing instance member rather than global struct
members reduces cache hit and improves performance
```

**Comment by liranoz12** [comment]
@OphirMunk Sounds great.

**Comment by OphirMunk** [comment]
PR accepted and merged

**Pull request closed without merging**